sql/executor.go

```diff
@@ -41,8 +41,9 @@ var errTransactionInProgress = errors.New("there is already a transaction in pro
 
 // An Executor executes SQL statements.
 type Executor struct {
-	db     client.DB
-	nodeID uint32
+	db      client.DB
+	nodeID  uint32
+	reCache *parser.RegexpCache
```

Comment:
I don't know about changing `RegexpCache.mu` to a `sync.Locker`: we don't want `RegexpCache` to be naively copied, and the `go vet` check for `sync.Mutex` helps us with that. This seems more valuable than what we'd gain by making `Executor.reCache` a value.